### ISSUE TITLE
material doesn't render subviewport texture

### ISSUE THREAD

**Issue by warriormaster12** [godotengine/godot#68162]
### Godot version

4.0-beta4

### System information

Operating System: Arch Linux KDE Plasma Version: 5.26.2 KDE Frameworks Version: 5.99.0 Qt Version: 5.15.7 Kernel Version: 6.0.6-arch1-1 (64-bit) Graphics Platform: Wayland Processors: 8 × AMD Ryzen 5 2500U with Radeon Vega Mobile Gfx Memory: 14,6 GiB of RAM Graphics Processor: AMD Radeon Vega 8 Graphics Manufacturer: HP Product Name: HP Laptop

### Issue description

Basically when you set material local to scene and then you add your viewport texture to the material. After that if you either play the scene or reopen the scene the material looses all of the information about subviewport resulting in a flat pink texture.

### Steps to reproduce

1. create a subviewport 
2. add somekind of a ui node as a child.
3. assign viewport to a material
4. play the scene or reopen the scene.

### Minimal reproduction project

[Uploading Viewport_texture_fix.tar.gz…]()

**Comment by warriormaster12** [comment]
[Viewport_texture_fix.zip](https://github.com/godotengine/godot/files/9918067/Viewport_texture_fix.zip)

The original link didn't for some reason so here's a working link.

**Comment by Calinou** [comment]
Is the viewport update mode set to **Always**?

**Comment by warriormaster12** [comment]
Yes. The error is that viewport width and height is 0 and when you set it manually in gdscript then it says that you need to set up the texture.

**Comment by RudyFisher7** [comment]
Thought I would pass this on here instead of opening a separate issue as it shows some more buggy behavior related to this. Been trying to use subviewport for UI in 3D world space using Sprite3D and stumbled on some issues when adjusting anchors and trying to get it to automatically expand with a V/HBoxContainer. I love Godot. Godot 4 is going to be a dream.
![subviewportbuggy](https://user-images.githubusercontent.com/91921130/200144226-046ca78e-a5d2-44a6-9264-90ee83cf6950.png)

**Comment by Rindbee** [comment]
This issue is similar to #68088, see https://github.com/godotengine/godot/issues/68088#issuecomment-1321067723 for the cause of this issue
